python/lsst/galaxy_shear/processShearTest.py

```diff
@@ -0,0 +1,144 @@
+        assert nGals * self.config.galaxyStampSize == imageBBox.getWidth()
+        n = imageBBox.getWidth() / nGals
+        self.dims = lsst.afw.geom.Extent2I(n, n)
+        offset = lsst.afw.geom.Extent2I(simCat[0][xKey], simCat[0][yKey])
+        for simRecord in simCat:
+            sourceRecord = sourceCat.addNew()
+            sourceRecord.setId(simRecord.get(idKey))
+            sourceRecord.set("psf_index", simRecord.get("psf_index"))
+            sourceRecord.set("psf_library", simRecord.get("psf_library"))
+            position = lsst.afw.geom.Point2I(simRecord.get(xKey), simRecord.get(yKey))
+            bbox = lsst.afw.geom.Box2I(position - offset, self.dims)
+            footprint = lsst.afw.detection.Footprint(bbox, imageBBox)
+            peakRecord = footprint.getPeaks().addNew()
+            peakRecord.setFx(position.getX())
+            peakRecord.setFy(position.getY())
+            sourceRecord.setFootprint(footprint)
+        return sourceCat
+
+    #   Run a measurement algorithm on the set of sources provided by
+    #   a GalSim catalog.  Also run a centroid algorithm if one is provided
+    def run(self, dataRef):
+        exposure = self.buildExposure(dataRef)
+        sourceCat = self.buildSourceCatalog(exposure.getBBox(lsst.afw.image.PARENT), dataRef)
+        images_file = dataRef.get("image", immediate=True)
+        #   the GalSim catalog will provide the number of the psf_library
+        #   and the index of the hdu with the psf for each source.
+        for source in sourceCat:
+            #   Locate the psb_library and get the indexed psf
+            dataRef.dataId["psf_index"] = source.get("psf_index")
+            dataRef.dataId["psf_library"] = source.get("psf_library")
+            psf_file = dataRef.get("psf_library", immediate=True)
+            data = psf_file.getArray().astype(numpy.float64)
+            kernel = lsst.afw.math.FixedKernel(lsst.afw.image.ImageD(data))
+            psf = lsst.meas.algorithms.KernelPsf(kernel)
+            # Create a bounding box around the galaxy image of self.dims
+            x = int(source.getFootprint().getCentroid().getX()+1) - (self.dims.getX()/2)
+            y = int(source.getFootprint().getCentroid().getY()+1) - (self.dims.getY()/2)
+            bbox = lsst.afw.geom.Box2I(lsst.afw.geom.Point2I(x,y), self.dims)
+            exp = lsst.afw.image.ExposureF(exposure, bbox)
+            exp.setPsf(psf)
+            #  Now do the measurements, calling the measure algorithms to increase speed
+            if self.centroidPlugin:
+                try:
+                    self.centroidPlugin.measure(source, exp)
+                except:
+                    self.centroidPlugin.fail(source)
+                    continue
+            if self.measPlugin:
+                try:
+                    self.measPlugin.measure(source, exp)
+                except:
+                    self.measPlugin.fail(source)
```

Comment:
You might want to catch the `Exception` objects and pass them to `fail()`, while handling `MeasurementError` and `FATAL_EXCEPTIONS` different, as `BaseMeasurementTask.callMeasure()` does.

It'd be nice if you could actually inherit from `BaseMeasurementTask` itself and call it directly, but I don't think that's worthwhile at this point, given that it would change a lot of changes to both `BaseMeasurementTask` and your code here.